cmd/gitops/add/profiles/cmd.go

```diff
@@ -0,0 +1,131 @@
+package profiles
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/go-resty/resty/v2"
+	"github.com/spf13/cobra"
+	"github.com/weaveworks/weave-gitops/cmd/internal"
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+	"github.com/weaveworks/weave-gitops/pkg/osys"
+	"github.com/weaveworks/weave-gitops/pkg/runner"
+	"github.com/weaveworks/weave-gitops/pkg/server"
+	"github.com/weaveworks/weave-gitops/pkg/services"
+	"github.com/weaveworks/weave-gitops/pkg/services/auth"
+	"github.com/weaveworks/weave-gitops/pkg/services/profiles"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var opts profiles.AddOptions
+
+// AddCommand provides support for adding a profile to a cluster.
+func AddCommand(client *resty.Client) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "profile",
+		Aliases:       []string{"profiles"},
+		Short:         "Add a profile to a cluster",
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Example: `
+		# Add a profile to a cluster
+		gitops add profile --name=podinfo --cluster=prod --version=1.0.0 --config-repo=ssh://git@github.com/owner/config-repo.git
+		`,
+		RunE: addProfileCmdRunE(client),
+	}
+
+	cmd.Flags().StringVar(&opts.Name, "name", "", "Name of the profile")
+	cmd.Flags().StringVar(&opts.Version, "version", "latest", "Version of the profile")
+	cmd.Flags().StringVar(&opts.ConfigRepo, "config-repo", "", "URL of external repository (if any) which will hold automation manifests")
+	cmd.Flags().StringVar(&opts.Cluster, "cluster", "", "Name of the cluster to add the profile to")
+	cmd.Flags().StringVar(&opts.Port, "port", server.DefaultPort, "Port the profiles API is running on")
+	cmd.Flags().BoolVar(&opts.AutoMerge, "auto-merge", false, "If set, 'gitops add profile' will merge automatically into the set --branch")
+
+	return cmd
+}
+
+func addProfileCmdRunE(client *resty.Client) func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		log := internal.NewCLILogger(os.Stdout)
+		fluxClient := flux.New(osys.New(), &runner.CLIRunner{})
+		factory := services.NewFactory(fluxClient, log)
+		providerClient := internal.NewGitProviderClient(os.Stdout, os.LookupEnv, auth.NewAuthCLIHandler, log)
+
+		if err := validateAddOptions(opts); err != nil {
+			return err
+		}
+
+		ns, err := cmd.Parent().Parent().Flags().GetString("namespace")
+		if err != nil {
+			return err
+		}
+		opts.Namespace = ns
+
+		config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(homedir.HomeDir(), ".kube", "config"))
+		if err != nil {
+			return fmt.Errorf("error initializing kubernetes config: %w", err)
+		}
+
+		clientSet, err := kubernetes.NewForConfig(config)
```

Comment:
They're different types unfortunately!

Reply:
Do we need to use the versioned clientset ?

Reply:
@bigkevmcd I've never used a versioned clientset before but from a quick search, it sounds like a good idea. However, since this could benefit a lot of `gitops`' `cmd`s (especially since most of them use `NewKubeHTTPClient` without making use of the `clientSet` that is returned), I suggest opening an issue to improve all of them rather than just `add profiles`!